DEVELOPMENT.adoc

```diff
@@ -0,0 +1,43 @@
+:uri-pkl-repo: https://github.com/apple/pkl
+
+= Pkl Go Development Guide
+
+== Debugging the Pkl Server
```

Comment:
@HT154 Does this work **generally** for debugging Pkl? I was just finding myself yesterday a bit annoyed at not being able to put breakpoints in Pkl sources.

Reply:
For debugging "regular" `pkl eval` invocations, you don't need to fuss with this, you can do it straight from intellij by finding the entrypoint (`pkl-cli/src/main/kotlin/org/pkl/cli/Main.kt`) and starting to run/debug from the `main` function. You can also modify the run configuration in that menu to pass CLI args, env vars, etc. as needed.

The approach documented here does work for general debugging as well, but it's required for debugging `pkl server` because the host process (eg. via pkl-go) currently must spawn the `pkl` process.

Reply:
That's very cool, thanks Josh!